cmd/dm-syncer/config.go

```diff
@@ -0,0 +1,383 @@
+
+	SkipDDLs []string `toml:"skip-ddls" json:"skip-ddls"`
+	// NOTE: SkipSQL and SkipEvents are no longer used, leave the comments to remind others.
+	// SkipSQLs is deprecated, please use SkipDDLs instead.
+	// SkipSQLs []string `toml:"skip-sqls" json:"-"` // omit it since it's deprecated
+	// SkipEvents is deprecated, please use SkipDMLs instead.
+	// SkipEvents []string   `toml:"skip-events" json:"-"` // omit it since it's deprecated
+	SkipDMLs []*SkipDML `toml:"skip-dmls" json:"skip-dmls"`
+
+	RouteRules []*RouteRule `toml:"route-rules" json:"route-rules"`
+
+	From config.DBConfig `toml:"from" json:"from"`
+	To   config.DBConfig `toml:"to" json:"to"`
+
+	EnableGTID  bool `toml:"enable-gtid" json:"enable-gtid"`
+	AutoFixGTID bool `toml:"auto-fix-gtid" json:"auto-fix-gtid"`
+
+	DisableCausality bool   `toml:"disable-detect" json:"disable-detect"`
+	SafeMode         bool   `toml:"safe-mode" json:"safe-mode"`
+	ConfigFile       string `json:"config-file"`
+
+	// NOTE: These four configs are all deprecated.
+	// We leave this items as comments to remind others there WERE old config items.
+	//topOnDDL               bool   `toml:"stop-on-ddl" json:"stop-on-ddl"`
+	//MaxDDLConnectionTimeout string `toml:"execute-ddl-timeout" json:"execute-ddl-timeout"`
+	//MaxDMLConnectionTimeout string `toml:"execute-dml-timeout" json:"execute-dml-timeout"`
+	//ExecutionQueueLength    int    `toml:"execute-queue-length" json:"execute-queue-length"`
+
+	EnableANSIQuotes bool           `toml:"enable-ansi-quotes" json:"enable-ansi-quotes"`
+	TimezoneStr      string         `toml:"timezone" json:"timezone"`
+	Timezone         *time.Location `json:"-"`
+
+	printVersion bool
+}
+
+// RouteRule is route rule that syncing
+// schema/table to specified schema/table
+// This config has been replaced by `router.TableRule`
+type RouteRule struct {
+	PatternSchema string `toml:"pattern-schema" json:"pattern-schema"`
+	PatternTable  string `toml:"pattern-table" json:"pattern-table"`
+	TargetSchema  string `toml:"target-schema" json:"target-schema"`
+	TargetTable   string `toml:"target-table" json:"target-table"`
+}
+
+// SkipDML defines config rule of skipping dml.
+// This config has been replaced by BinLog filter.
+type SkipDML struct {
+	Schema string `toml:"db-name" json:"db-name"`
+	Table  string `toml:"tbl-name" json:"tbl-name"`
+	Type   string `toml:"type" json:"type"`
+}
+
+func loadMetaFile(metaFile string) (*config.Meta, error) {
+	file, err := os.Open(metaFile)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	defer file.Close()
+
+	meta := &config.Meta{}
+	_, err = toml.DecodeReader(file, meta)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return meta, nil
+}
+
+func (oc *oldConfig) convertToNewFormat() (*config.SubTaskConfig, error) {
+	meta, err := loadMetaFile(oc.Meta)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	newTask := &config.SubTaskConfig{
+		Name: "dm-syncer",
```

Comment:
Syncer unit in DM uses the downstream database to record checkpoint, and this `Name` will be used as the prefix of the checkpoint table name, so is it better to export this config item in `newSubTaskConfig`?

Reply:
ok, the config name is `cp-table-prefix`